ui/components/app/confirm/sender-info/sender-info.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { memo, useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+import {
+  BackgroundColor,
+  BorderRadius,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../selectors';
+import { Box } from '../../../component-library';
+import {
+  ConfirmInfo,
+  ConfirmInfoRowConfig,
+  ConfirmInfoRowType,
+} from '../info/info';
+
+const SenderInfo: React.FC = memo(() => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+  const [rowConfigs, setRowConfigs] = useState<ConfirmInfoRowConfig[]>([]);
+
+  useEffect(() => {
+    if (
+      !currentConfirmation ||
+      currentConfirmation.type !== MESSAGE_TYPE.PERSONAL_SIGN ||
+      !currentConfirmation.msgParams?.origin
+    ) {
+      return;
+    }
+    setRowConfigs([
+      {
+        label: t('origin'),
+        type: ConfirmInfoRowType.UrlType,
+        rowProps: {
+          url: currentConfirmation.msgParams?.origin,
+        },
+      },
+    ]);
+  }, [currentConfirmation]);
+
+  const args = useMemo(() => ({ rowConfigs: [...rowConfigs] }), [rowConfigs]);
+
+  if (!rowConfigs?.length) {
+    return null;
+  }
+
+  return (
+    <Box
+      backgroundColor={BackgroundColor.backgroundDefault}
+      borderRadius={BorderRadius.MD}
+      padding={2}
+      marginBottom={4}
+    >
+      <ConfirmInfo {...args} />
+    </Box>
+  );
+});
+
+export default SenderInfo;
```

Comment:
I'm struggling a bit to see the bigger picture. How will the file paths and content scale for transaction types and additional conditions? 

e.g. I see this design that appears to include the sender and is beyond the scope of sender-info 
<img width="418" alt="Screenshot 2024-02-07 at 12 37 19 AM" src="https://github.com/MetaMask/metamask-extension/assets/20778143/b3e7818f-d1c9-448d-8a5a-423cf1249d76">

Reply:
@digiwand : my plan is to add more fields here as we develop those pages.

Reply:
Hey @digiwand : I updated the PR tp create more generic `info` component.